waku/protocol/v2/waku_filter.nim

```diff
@@ -130,3 +113,7 @@ proc subscription*(proto: WakuFilter): MessageNotificationSubscription =
     await allFutures(futures)
 
   MessageNotificationSubscription.init(@[], handle)
+
+proc filter*(proto: WakuFilter, handler: proc(msgs: MessagePush): Future[void]) =
```

Comment:
I'm not 100% sure I understand this signature and how it is meant to be used

Reply:
Similar to all other functions that have a handler method. I just haven't declared a type yet.